app/src/main/java/org/mozilla/fenix/search/SearchFragment.kt

```diff
@@ -288,6 +350,27 @@ class SearchFragment : Fragment(), BackHandler {
         } else null
     }
 
+    private fun updateSearchSuggestionsHintVisibility(state: SearchFragmentState) {
+        if (state.showSearchSuggestionsHint) {
+            showSearchSuggestionsHint()
```

Comment:
Since the helper functions are very similar, they should be collapsed into a single function. Kotlin view binding and `isVisible` can make this even shorter.

```kotlin
private fun updateSearchSuggestionsHintVisibility(state: SearchFragmentState) {
  view?.search_suggestions_onboarding?.isVisible = state.showSearchSuggestionsHint
}
```

Reply:
I was using them separately to make it obvious that when showing the hint, the target view could be a `ViewStub` (setting visible on a stub also inflates it). I've collapsed them into a single function, but I kept the explicit `findViewById`, because synthetic import does not play well with viewstub. (it keeps the type of `ViewStub`, when actually the `ViewStub` becomes a `View`.)